spec/requests/relationships_spec.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Relationships', type: :request do
+  describe 'create' do
+    context 'when user is authenticated' do
+      let!(:users) { create_list(:user, 2) }
+
+      before do
+        sign_in users[0]
+      end
+
+      it 'returns HTTP success' do
+        get show_user_path(users[1].nickname)
+        expect(response).to be_successful
+      end
+
+      it 'follows another user (through AJAX)' do
+        expect do
+          post relationships_path, params: { followed_id: users[1].id }, xhr: true
+        end.to change(Relationship, :count).by(1)
+      end
+
+      it 'responds with JavaScript' do
+        post relationships_path, params: { followed_id: users[1].id }, xhr: true
+        expect(response.content_type).to eq('text/javascript; charset=utf-8')
+        expect(response.body).to include('Unfollow')
+      end
+
+      it 'follows another user (through HTML)' do
+        expect do
+          post relationships_path, params: { followed_id: users[1].id }
+        end.to change(Relationship, :count).by(1)
+      end
+
+      it 'responds with HTML' do
+        post relationships_path, params: { followed_id: users[1].id }
+        expect(response).to redirect_to(show_user_path(users[1].nickname))
+        follow_redirect!
+        expect(response.body).to include('Unfollow')
+      end
+    end
+
+    context 'when user is not authenticated' do
+      let(:user) { create(:user) }
+
+      it 'redirects to sign in' do
+        get show_user_path(user.nickname)
+        expect(response).to redirect_to(user_session_path)
+      end
```

Comment:
This context should test that the creation is not allowed when user is not authenticated